plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -494,7 +496,7 @@ public Stream<TableColumnsMetadata> streamTableColumns(ConnectorSession session,
                 // intentionally skip case when table snapshot is present but it lacks metadata portion
                 return metastore.getMetadata(metastore.getSnapshot(table, session), session).stream().map(metadata -> {
                     List<ColumnMetadata> columnMetadata = getColumns(metadata).stream()
-                            .map(DeltaLakeMetadata::getColumnMetadata)
+                            .map(column -> getColumnMetadata(column, getColumnComments(metadata)))
```

Comment:
`getColumnComments(tableHandle.getMetadataEntry())` is performed once for every column, should be only once